internal/provider/cached_image_resource.go

```diff
@@ -293,128 +295,246 @@ func (r *CachedImageResource) Configure(ctx context.Context, req resource.Config
 	r.client = client
 }
 
-// setComputedEnv sets data.Env and data.EnvMap based on the values of the
-// other fields in the model.
-func (data *CachedImageResourceModel) setComputedEnv(ctx context.Context) diag.Diagnostics {
-	env := make(map[string]string)
+func optionsFromDataModel(data CachedImageResourceModel) (eboptions.Options, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	var opts eboptions.Options
 
-	env["ENVBUILDER_CACHE_REPO"] = tfValueToString(data.CacheRepo)
-	env["ENVBUILDER_GIT_URL"] = tfValueToString(data.GitURL)
+	// Required options. Cannot be overridden by extra_env.
+	opts.CacheRepo = data.CacheRepo.ValueString()
+	opts.GitURL = data.GitURL.ValueString()
+
+	// Other options can be overridden by extra_env, with a warning.
+	// Keep track of which options are overridden.
+	overrides := make(map[string]struct{})
 
 	if !data.BaseImageCacheDir.IsNull() {
-		env["ENVBUILDER_BASE_IMAGE_CACHE_DIR"] = tfValueToString(data.BaseImageCacheDir)
+		overrides["ENVBUILDER_BASE_IMAGE_CACHE_DIR"] = struct{}{}
+		opts.BaseImageCacheDir = data.BaseImageCacheDir.ValueString()
```

Comment:
nit: I understand there is no simple way to replace this _wall of condition-override-value 's_ with any dynamic logic?

Reply:
One option I've been considering is using struct tags to match up the fields between these two. A 'dirty hack' could be using `json.Marshal` / `json.Unmarshal` to convert between the two. Some of the subtleties of the Terraform data model might be lost though (e.g. "unknown" versus "zero value").

Reply:
If there is a chance of losing the information, then let's leave it as is, although I admit it is tempting to make it more dynamic.